src/app/services/teachers.service.ts

```diff
@@ -1,81 +1,109 @@
+import { AbstractControl, FormGroup } from '@angular/forms';
 import { editTeacher } from './../store/teachers/teachers.action';
 import { NotificationService } from './notification.service';
 import { environment } from './../../environments/environment';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { teacherAction, addOneTeacher } from '../store/teachers/teachers.action';
-import { Observable, Observer, Subject } from 'rxjs';
+import {
+  teacherAction,
+  addOneTeacher
+} from '../store/teachers/teachers.action';
+import { Subject } from 'rxjs';
+import { HttpGetReponse, HttpPostPutReponse } from '../models/HttpResponse.model';
+import { Teacher } from '../models/teacher.model';
 
 
 @Injectable({
   providedIn: 'root'
 })
 export class TeachersService {
-
   public subject = new Subject<string | ArrayBuffer>();
   private BASE_URI = environment.APIEndpoint;
 
-
-  constructor(private http: HttpClient,
-              private store: Store<{teachers}>,
-              private notify: NotificationService) {
-  }
+  constructor(
+    private http: HttpClient,
+    private store: Store<{ teachers }>,
+    private notify: NotificationService
+  ) {}
 
   getTeachers() {
-    return this.http.get(`${this.BASE_URI}teachers`)
-    .subscribe(response => {
-      // @ts-ignore
-      this.store.dispatch(teacherAction({ teachersList: response.data }));
-    });
+    return this.http
+      .get(`${this.BASE_URI}teachers`)
+      .subscribe((response: HttpGetReponse) => {
+        this.store.dispatch(teacherAction({ teachersList: response.data }));
+      });
   }
 
-  editTeacher(teacherId, data) {
-    return this.http.put(`${this.BASE_URI}admin/teachers/${teacherId}`, data, {
```

Comment:
worth to create const for strings like `admin/teachers/${teacherId}` and add somewhere method eg `normalizeUrl/getUrl` in which will return correct full URL. Might be done in a separate task